classes/TaskRunner/Miscellaneous/UpdateConfig.php

```diff
@@ -95,12 +95,23 @@ public function run()
             }
 
             $xmlFile = $configurationData['archive_xml'];
-            if (!empty($xmlFile) && !file_exists($this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH) . DIRECTORY_SEPARATOR . $xmlFile)) {
+            $fullXmlPath = $this->container->getProperty(UpgradeContainer::DOWNLOAD_PATH) . DIRECTORY_SEPARATOR . $xmlFile;
+            if (!empty($xmlFile) && !file_exists($fullXmlPath)) {
                 $this->error = true;
                 $this->logger->info($this->translator->trans('File %s does not exist. Unable to select that channel.', [$xmlFile]));
 
                 return false;
             }
+
+            $xmlVersion = $this->getXmlVersion($fullXmlPath);
+
+            if ($xmlVersion != $targetVersion) {
```

Comment:
```suggestion
            if ($xmlVersion !== $targetVersion) {
```

Reply:
extractPrestashopVersionFromZip return a string, same for getXmlVersion.

Reply:
Than having the same type is a solid reason to have a strict check, and disallow any type casting.